src/System.Slices/System/Buffers/OwnedMemory.cs

```diff
@@ -178,16 +178,25 @@ bool IMemory<T>.TryGetArray(long id, out ArraySegment<T> buffer)
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        internal Span<T> GetSpanInternal(long id)
+        internal Span<T> GetSpanInternal(long id, int index, int length)
         {
             VerifyId(id);
-            return Span;
+            var array = Array;
+            if (array != null)
+                return new Span<T>(array, Offset + index, length);
+            else unsafe
+                {
+                    if (length > Length - index) throw new ArgumentOutOfRangeException();
+                    var offsetBytes = Unsafe.SizeOf<T>() * index;
```

Comment:
Are you sure? The check above ensures that either index is smaller than Length or length is negative. if length is negative, the next line (ctor) will fail. If index is smaller than Length, then the multiplication cannot overflow or we will have much bigger problems.  

Having said that, I just stole the implementation from System.Memory.Span<T>.Slice. Though I am not sure all the checks are necessary (as mentioned above).